### PR TITLE
 client, templates: Fix ban post status

### LINKED ISSUE #857: Ban HTML broken on live updates
`<b class="admin banned" banned="">USER WAS BANNED FOR THIS POST BY "admin" FOR 1 MINUTE: k</b>`

### PR DESCRIPTION
**Pull request by Chiiruno**
Fixes #857
Do you see anything else that might be a problem?
I can't exactly test whether or not the ban itself shows up live for others.

### CODE REVIEW

**Comment by Chiiruno** [comment]
Is it possible that the modlog message is somehow being read after the ban/delete/meidovision message sometimes, even though we send it first?
Perhaps a loop until the post modlog is populated?

**Comment by Chiiruno** [comment]
Anyway, if that is the case, we could remove the length and type check in the client.

**Comment by Chiiruno** [comment]
This is caused by the render loop, just need a while loop, and ensure renderStatus is async.
Don't merge yet.

**Comment by Chiiruno** [comment]
Actually, I could probably just return and it would be fine.

**Comment by Chiiruno** [comment]
@bakape All good?

**Comment by Chiiruno** [comment]
Okay, this should ensure the render loop doesn't overwrite the handler's call if log doesn't exist yet.

**Comment by bakape** [comment]
Test this by changing the IP of the post with psql.

**Comment by Chiiruno** [comment]
It just shows without the log info, just the default `lang` value.
It does update whenever say, meido vision or the post is deleted.
Not really sure why this happens, thoughts?

**Comment by Chiiruno** [comment]
What's really weird is that this doesn't happen with deletion or meido vision.

**Comment by Chiiruno** [comment]
I know why ban isn't working now, it doesn't use moderatepost.
I just need to add it to bans.go.

File: go/src/meguca/db/bans.go [hunk 1]
```diff
@@ -144,9 +144,21 @@ func Ban(board, reason, by string, expires time.Time, log bool, ids ...uint64) (
 
 	if !IsTest {
 		for _, post := range posts {
+			mLog, err := GetPostModLog(post.id)
+
+			if err != nil {
+				return err
+			}
+
+			err = auth.ModLogPost(post.id, post.op, mLog)
```
Comment (bakape) [inline go/src/meguca/db/bans.go]: What is this for?

Comment (Chiiruno) [reply go/src/meguca/db/bans.go]: https://github.com/bakape/meguca/blob/master/go/src/meguca/db/admin.go#L77-L92
https://github.com/bakape/meguca/blob/master/go/src/meguca/websockets/feeds/feed.go#L21
https://github.com/bakape/meguca/blob/master/go/src/meguca/websockets/feeds/feed.go#L311-L313
https://github.com/bakape/meguca/blob/master/go/src/meguca/websockets/feeds/feeds.go#L27
https://github.com/bakape/meguca/blob/master/go/src/meguca/websockets/feeds/feeds.go#L303-L318
It's the whole reason this works.

Comment (bakape) [reply go/src/meguca/db/bans.go]: Too roundabout and leads to messier and harder to follow code. Could you centralise writing mod log to posts by moving it to `logModderation()`?

Comment (Chiiruno) [reply go/src/meguca/db/bans.go]: We're not writing mod-log to posts server-side, we are sending mod-log info pertaining to a post to the client via websockets.
What I can do is change auth.ModLogPost to common.ModLogPost, remove the mLog input, and have it fetch from the db from websockets instead.

**Comment by bakape** [comment]
Yes, but the whole `write mod log to db -> fetch mod log from db ->
propagate to feeds` chain has an extra DB read. I'd prefer to avoid that.

**Comment by Chiiruno** [comment]
Oh, I see what you mean now.
Okay, no problem, in that case I'll keep auth.ModLogPost, since we'll have the log from logModeration.

**Comment by bakape** [comment]
Preferably I'd like to abstract that away and have that done entirely in
logModeration(). It will also make integrating any future moderation
functionality easier.

**Comment by Chiiruno** [comment]
To do that, we would have to move logModeration from the db to websockets, which isn't a problem.
The other way around would result in circular imports.

**Comment by Chiiruno** [comment]
@bakape This is going to take a somewhat large rewrite, if you want to merge this now for the fix, I'll open up a PR in a few days, if not sooner, containing the rewrite.

**Comment by bakape** [comment]
I'll merge this now. Please submit the rewrite later.

**Comment by Chiiruno** [comment]
Yes sir.

**Pull request merged**